File: backend/BeautyCareClinic.Domain/Entities/Payment.cs
namespace BeautyCareClinic.Domain.Entities;

public class Payment
{
    public Guid Id { get; set; }
    public Guid OrderId { get; set; }
    public decimal Amount { get; set; }
    public string PaymentMethod { get; set; } = string.Empty;
    public DateTime PaymentDate { get; set; }
    public Guid RecordedByUserId { get; set; }
    public string RecordedByFullName { get; set; } = string.Empty;

    public CustomerOrder Order { get; set; } = null!;
}

File: frontend/src/types/Payment.ts
export interface Payment {
  id: string;
  // orderId is the canonical field from the API; customerOrderId is the legacy mock-data alias
  orderId?: string;
  customerOrderId?: string;
  amount: string | number;
  paymentMethod?: string;
  // Legacy mock-data field
  method?: string;
  paymentDate: string;
  recordedByUserId?: string;
  recordedByFullName?: string;
  // Legacy mock-data fields
  createdDate?: string;
  createdByUserId?: string;
}
